PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -624,8 +624,13 @@ StartBooting (
       DEBUG ((DEBUG_ERROR, "EntryPoint is not found\n"));
       return RETURN_INVALID_PARAMETER;
     }
-    BeforeOSJump ("Starting MB Kernel ...");
-    JumpToMultibootOs((IA32_BOOT_STATE*)&MultiBoot->BootState);
+    if (FeaturePcdGet (PcdPreOsCheckerEnabled) && IsPreOsCheckerLoaded ()) {

```

Comment:
For PreOsChecker, will it boot into LOADED_IMAGE_LINUX or LOADED_IMAGE_MULTIBOOT?
I saw it was checked in both path.

Reply:
The pre-OS checker/payload being used would need to support it; someone else was able to validate their pre-OS checker binary is working as expected with this multiboot flow.

Reply:
Request was to enable PreOsChecker even for MB ELF images to support for example ACRN boot.

Reply:
Got it.